wagtail/snippets/models.py

```diff
@@ -69,35 +69,60 @@ def get_edit_url(self, instance):
             )
 
 
-def register_snippet(model, viewset=None):
+def register_snippet(registerable, viewset=None):
     if DEFER_REGISTRATION:
         # Models may not have been fully loaded yet, so defer registration until they are -
         # add it to the list of registrations to be processed by register_deferred_snippets
-        DEFERRED_REGISTRATIONS.append((model, viewset))
+        DEFERRED_REGISTRATIONS.append((registerable, viewset))
     else:
-        _register_snippet_immediately(model, viewset)
+        _register_snippet_immediately(registerable, viewset)
 
-    return model
+    return registerable
 
 
-def _register_snippet_immediately(model, viewset=None):
+def _register_snippet_immediately(registerable, viewset=None):
     # Register the viewset and formfield for this snippet model,
     # skipping the check for whether models are loaded
 
-    if model in SNIPPET_MODELS:
-        # Do not create duplicate registrations of the same model
+    from wagtail.snippets.views.snippets import SnippetViewSet, SnippetViewSetGroup
+
+    if isinstance(registerable, str):
+        registerable = import_string(registerable)
```

Comment:
Nice to have, but I'm happy to remove this if we'd rather require this to be the actual class